### PR TITLE
Remove unused libraries

### PR DESCRIPTION
**Pull request by thealmarty**
This removes LLVM, arithmetic circuit, and interaction net, which were added to facilitate future plans. Plans have changed and these won't be needed for a while. They are removed for now to keep Juvix lighter.

### CODE REVIEW

**Comment by thealmarty** [comment]
This compiles on my laptop but fails on the CI. Looking into it.

**Comment by thealmarty** [comment]
The latest commit compiles on both my laptop and desktop. I think the CI failures are from caching? @cwgoes any idea?

File: library/Core/src/Juvix/Core/IR/Types/Base.hs [hunk 1]
```diff
@@ -63,7 +63,7 @@ extensible
         Elim (Elim primTy primVal)
       deriving (Eq, Show, Generic, Data, NFData)
 
-    -- | inferable terms
+    -- inferable terms
```
Comment (thealmarty) [inline library/Core/src/Juvix/Core/IR/Types/Base.hs:66]: Haddock doc not yet supported in template haskell. Removing for now.

**Comment by cwgoes** [comment]
> The latest commit compiles on both my laptop and desktop. I think the CI failures are from caching? @cwgoes any idea?

Looks like `arithmetic-circuit` is still listed as a dependency in `package.yaml` - probably will need to remove that.

**Comment by thealmarty** [comment]
> > The latest commit compiles on both my laptop and desktop. I think the CI failures are from caching? @cwgoes any idea?
> 
> Looks like `arithmetic-circuit` is still listed as a dependency in `package.yaml` - probably will need to remove that.

Oops you're right. Thanks. I guess `stack clean` isn't sufficient! I'm checking with a `/tmp` dir now.

**Comment by thealmarty** [comment]
Still working on it. Not ready for review yet.

**Comment by thealmarty** [comment]
Per meeting pausing this PR to work on other more important issues.

**Comment by thealmarty** [comment]
- cleaned up some warnings.
- removed LLVM/INet/ArithmeticCircuit libraries.
- moved EAC out of core.

**Comment by cwgoes** [comment]
@thealmarty I think you might have some leftover local files since the org-gen is different. Fixed quickly.

**Review by cwgoes — APPROVED** [review]
ACK

**Pull request merged**